src/Components/CreateImageWizard/formComponents/RepositoryUnavailable.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import { Alert, Button } from '@patternfly/react-core';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons';
+
+import { useCheckRepositoriesAvailability } from '../../../Utilities/checkRepositoriesAvailability';
+import { useGetEnvironment } from '../../../Utilities/useGetEnvironment';
+
+const RepositoryUnavailable = () => {
+  const { isBeta } = useGetEnvironment();
+
+  if (useCheckRepositoriesAvailability()) {
```

Comment:
There might be an issue there, because `useCheckRepositoriesAvailability` only fetches the 100 first repos for x86:

```
  const { data, isSuccess } = useListRepositoriesQuery({
    availableForArch: 'x86_64',
    availableForVersion: version,
    limit: 100,
    offset: 0,
  });
```

So if there is a lot of repos, we could endip missing some unavailable repos.

Reply:
Oof, you're right. I'm wondering what the best solution would be. The limit is 100 by default so either we can set an artificial higher number that we hope wouldn't be exceeded... or we do the whole repo shebang with initial request, followed up by a request with the count of the repos from the initial one :thinking:

Reply:
For now chose the first route with a to do note

Reply:
I think we should keep a low page count but ask for next pages if needed. You can also mix that with the early return, so you check the first page, if something unavailable is found in it, you return true immediately, if not you continue checking the next page.

Reply:
Can't think of a way to check by page and not use a hook in a cycle. Re-used a part of a code we already utilise in `Repositories.js`. Making an initial request just to get the total count of the repositories and if that is higher than 100 making a follow up request with limit set to count to fetch all of them. `data` and `isSuccess` is then populated based on the count.

Will need to invest in AC for next year, my brain is melting...

Reply:
Nice pair programming session :rocket: